pom.xml

```diff
@@ -355,6 +342,20 @@
 
     <pluginManagement>
       <plugins>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <version>${maven-surefire-plugin.version}</version>
+          <configuration>
+            <rerunFailingTestsCount>3</rerunFailingTestsCount>
```

Comment:
unfortunately the website says only for junit 4.x
https://maven.apache.org/surefire/maven-surefire-plugin/examples/rerun-failing-tests.html
have you tried it and seen it working?

Reply:
Sure thing, it is supported in 3.0.0-M4
![image](https://user-images.githubusercontent.com/69956021/146864524-7cf66cc4-a3d8-4ed6-bc96-76d4e6483baa.png)

It works fine as `RUN 1 failed` and `RUN 2 PASS`

Reply:
Mentioned here https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html
```
(JUnit 4+ providers and JUnit 5+ providers since 3.0.0-M4) The number of times each failing test will be rerun. If set larger than 0, rerun failing tests immediately after they fail. If a failing test passes in any of those reruns, it will be marked as pass and reported as a "flake". However, all the failing attempts will be recorded.
Default value is: 0.
User property is: surefire.rerunFailingTestsCount.
```

Reply:
cool